### PR TITLE
chore: migrate module keyword to namespace

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Overview

Ref: https://github.com/microsoft/TypeScript/issues/51825

Enables [`@typescript-eslint/prefer-namespace-keyword`](https://typescript-eslint.io/rules/prefer-namespace-keyword) in this repo. The steps I took were pretty straightforward:

1. Replaced the `rules` object in `.eslintrc.cjs` with `{ '@typescript-eslint/prefer-namespace-keyword': 'error' }`
2. Replaced `eslint-` in comments with `JOSH_IGNORE_-` _(so that ESLint wouldn't error on missing rules)_
3. Ran `npx eslint "./types/**/*.ts" "./types/**/*.tsx" --fix`
4. Undid the `JOSH_IGNORE_-` replacements
5. Reset `.eslintrc.cjs` back to `master`, then added the `'@typescript-eslint/prefer-namespace-keyword': 'error'` at the end of `rules`

### CODE REVIEW

File: types/hapi__nes/index.d.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ declare module '@hapi/hapi' {
     }
 }
 
-declare module nes {
+declare namespace nes {
```
Comment (JoshuaKGoldberg) [inline types/hapi__nes/index.d.ts:45]: Looks like this module has a compile error in TypeScript 4.8 but not 4.9, unrelated to this change. https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/5767649258/job/15637662697?pr=66292

<details>
<summary>Errors copied from CI</summary>

```plaintext
Error in hapi__nes
Error: /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/broadcast-server.ts:6:20
ERROR: 6:20   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.

/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/hapi__nes-tests.ts:4:20
ERROR: 4:20   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.
ERROR: 23:15  expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
Parameter 'err' implicitly has an 'any' type.

/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/route-authentication-server.ts:8:20
ERROR: 8:20   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.

/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/route-invocation-server.ts:6:20
ERROR: 6:20   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.

/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/subscription-filter-server.ts:8:18
ERROR: 8:18   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.
ERROR: 48:33  expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
Parameter 'options' implicitly has an 'any' type.

/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hapi__nes/test/subscriptions-server.ts:6:18
ERROR: 6:18   expect  Compile error in typescript@4.8 but not in typescript@4.9.
Fix with a comment '// Minimum TypeScript Version: 4.9' just under the header.
'Server' only refers to a type, but is being used as a value here.
```

</details>

Comment (peterblazejewicz) [reply types/hapi__nes/index.d.ts:45]: this one is not very active (native module), the source on DT can be removed (to skip investigation and working on fixing the types), I could do this in parallel

**Review by ElizabethSamuel-MSFT — APPROVED** [review]
signing off for office-js and office-js-preview - our downstream processes and documentation aren't adversely affected

File: types/hapi__nes/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/hapi__nes/index.d.ts:45]: PR: let's give the owners a chance to make an opinion:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66372

**Review by sandersn — APPROVED** [review]

File: types/hapi__nes/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/hapi__nes/index.d.ts:45]: amended in master branch

**Review by Rick-Kirkham — APPROVED** [review]

File: types/hapi__nes/index.d.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply types/hapi__nes/index.d.ts]: Thanks! Merged from `master`. 👍

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## db.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/db.js@latest/))
was missing the following properties:
1. version
## doctrine ([<kbd>unpkg</kbd>](https://unpkg.com/browse/doctrine@latest/))
was missing the following properties:
1. Error
## lokijs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/lokijs@latest/))
was missing the following properties:
1. deepFreeze
1. freeze
1. unFreeze
1. Resultset
1. Comparators
## murmurhash3js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/murmurhash3js@latest/))
was missing the following properties:
1. version
## nes ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nes@latest/))
was missing the following properties:
1. plugin
## node-hue-api ([<kbd>unpkg</kbd>](https://unpkg.com/browse/node-hue-api@latest/))
was missing the following properties:
1. BridgeApi
1. api
1. connect
1. scheduledEvent
1. scene

<details>
<summary>as well as these 3 other properties...</summary>
<p>timer, locateBridges, searchForBridges</p>
</details>

## precise ([<kbd>unpkg</kbd>](https://unpkg.com/browse/precise@latest/))
was missing the following properties:
1. precise
## revalidator ([<kbd>unpkg</kbd>](https://unpkg.com/browse/revalidator@latest/))
was missing the following properties:
1. mixin
## simple-url-cache ([<kbd>unpkg</kbd>](https://unpkg.com/browse/simple-url-cache@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'simple-url-cache'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. pool
1. locks
1. helpers
1. StorageInstance

<details>
<summary>as well as these 1 other properties...</summary>
<p>RedisPool</p>
</details>

## swiz ([<kbd>unpkg</kbd>](https://unpkg.com/browse/swiz@latest/))
was missing the following properties:
1. help
1. check
1. checkPartial
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 734ad381fef0f719ace0a9a65572e50e0cfb4fc9
</p>

File: README.md [hunk 2]
```diff
@@ -412,7 +412,7 @@ Once a week the Definition Owners are synced to the file [.github/CODEOWNERS](ht
 
 ## The history of Definitely Typed
 
-Definitely Typed is one of the most active repositories on GitHub. You might have wondered how the project came to be. A history of Definitely Typed exists, that was put together by @johnnyreilly. It tells the story of the early days of Definitely Typed, from a repository created by @borisyankov, to the point where it became a pivotal part of the TypeScript ecosystem. [You can read the story of Definitely Typed here](https://johnnyreilly.com/definitely-typed-the-movie).  
+Definitely Typed is one of the most active repositories on GitHub. You might have wondered how the project came to be. A history of Definitely Typed exists, that was put together by @johnnyreilly. It tells the story of the early days of Definitely Typed, from a repository created by @borisyankov, to the point where it became a pivotal part of the TypeScript ecosystem. [You can read the story of Definitely Typed here](https://johnnyreilly.com/definitely-typed-the-movie).
```
Comment (JoshuaKGoldberg) [inline README.md:415]: Not an intentional part of this PR, but I like the change. 😄

**Review by peterblazejewicz — APPROVED** [review]

File: .eslintrc.cjs [hunk 3]
```diff
@@ -139,6 +139,7 @@ module.exports = {
                 'ts-check': false,
             }
         ],
+        '@typescript-eslint/prefer-namespace-keyword': 'error',
```
Comment (sandersn) [inline .eslintrc.cjs:142]: this change now needs to be made on the microsoft/definitelytyped-tools repo in the eslint-plugin package.

**Pull request merged**